hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/TrashPolicyOzone.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.fs.ozone;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.TrashPolicyDefault;
+import org.apache.hadoop.fs.FileAlreadyExistsException;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.fs.InvalidPathException;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.ozone.OFSPath;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.apache.hadoop.ozone.om.helpers.OzoneFSUtils;
+import org.apache.hadoop.util.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+
+/**
+ * TrashPolicy for Ozone Specific Trash Operations.
+ */
+public class TrashPolicyOzone extends TrashPolicyDefault {
```

Comment:
Is this the same as TrashPolicyOzone.java under ozone-manager module?

Reply:
Yes it is same as ozone-manager module.
The problem is when client changes trash policy to use ozone trash policy, it can't find TrashPolicyOzone as it's not present inside ozone fs client jar.
For workaround we have to put ozone-manager service jar to make it work.

Reply:
@adoroszlai thanks for the suggestion, let me try.

Reply:
**Before fix:** 
Perform delete and use OzoneTrashPolicy 
`  hdfs dfs -Dfs.trash.classname=org.apache.hadoop.fs.ozone.TrashPolicyOzone -rm ofs://om/vol1/bucket1/key1
`
```
  24/11/20 06:40:12 DEBUG fs.FsShell: Error
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.ozone.TrashPolicyOzone not found
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2734)
	at org.apache.hadoop.fs.TrashPolicy.getInstance(TrashPolicy.java:145)
	at org.apache.hadoop.fs.Trash.<init>(Trash.java:58)
	at org.apache.hadoop.fs.Trash.moveToAppropriateTrash(Trash.java:95)
	at org.apache.hadoop.fs.shell.Delete$Rm.moveToTrash(Delete.java:153)
	at org.apache.hadoop.fs.shell.Delete$Rm.processPath(Delete.java:118)
	at org.apache.hadoop.fs.shell.Command.processPathInternal(Command.java:371)
	at org.apache.hadoop.fs.shell.Command.processPaths(Command.java:334)
	at org.apache.hadoop.fs.shell.Command.processPathArgument(Command.java:307)
	at org.apache.hadoop.fs.shell.Command.processArgument(Command.java:289)
	at org.apache.hadoop.fs.shell.Command.processArguments(Command.java:273)
	at org.apache.hadoop.fs.shell.FsCommand.processRawArguments(FsCommand.java:120)
	at org.apache.hadoop.fs.shell.Command.run(Command.java:180)
	at org.apache.hadoop.fs.FsShell.run(FsShell.java:328)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:81)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:95)
	at org.apache.hadoop.fs.FsShell.main(FsShell.java:391)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.ozone.TrashPolicyOzone not found
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2702)
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2726)
	... 16 more
Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.ozone.TrashPolicyOzone not found
	at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2606)
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2700)
	... 17 more
-rm: Fatal internal error
```


**After fix:**
`24/11/20 08:15:37 INFO ozone.TrashPolicyOzone: Moved: 'ofs://om/vol1/bucket1/key1' to trash at: ofs://om/vol11/bucket1/.Trash/systest/Current/key1`

Reply:
Thanks @ashishkumar50 for updating the patch.  Sorry about the findbugs failure.  Since the client-side class is new, I guess we can rename it to avoid the problem.

`org.apache.hadoop.fs.ozone.TrashPolicyOzone -> org.apache.hadoop.fs.ozone.OzoneTrashPolicy`

Reply:
Thanks @adoroszlai, Renamed class.